pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -871,14 +870,6 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             offload_optimizer_device="nvme",
         )
 
-    @property
-    def checkpoint_io(self) -> CheckpointIO:
-        return self._checkpoint_io
-
-    @checkpoint_io.setter
-    def checkpoint_io(self, plugin: CheckpointIO) -> None:
-        raise MisconfigurationException("DeepSpeed currently does not support custom checkpoint plugins.")
```

Comment:
#8743 first implemented a `DeepSpeedCheckpointPlugin` which was removed before merge, so this is not necessary

Reply:
if someones does this:
```py
plugin = DeepSpeedPlugin()
plugin.checkpoint_io = CustomCheckpointIO()
```
this should still raise this exception?

Reply:
Why? This class does not have any custom checkpointing logic

Reply:
Mmmm I'm conflicted here.

This plugin doesn't use a checkpoint_io at all. Which might suggest it's good to raise

But the base implementation will set (through the setter) a default `checkpoint_io` which means the exception would always be raised.

I think it's more natural not to override the setter and leave it be. The problem is that this TTP implementation violates the base interface as the base requires the `checkpoint_io` but this subclass does not.

Reply:
if we remove this and someone calls plugin.checkpoint_io, it will return `TorchCheckpointIO` I think.
To avoid the conflict with the base one you can check
```py
def checkpoint_io(self, plugin: CheckpointIO)
    if plugin is not None:
        raise MisconfigurationException(...)
```
since it's optional anyways in the base class.

Reply:
Done, although I still think we shouln't have these checks. They limit the flexibiltiy.